### PR TITLE
Update logging and terraform, and default to secure-HTTPS

### PR DESCRIPTION
**Pull request by alexdunnjpl**
## 🗒️ Summary
Various fixes to logging and terraform labeling.  Switches default behaviour to check host certs, and adds `DEV_MODE` env var to disable

## ⚙️ Test Data and/or Report
Unit tests pass.
Terraform manually tested

## ♻️ Related Issues
Related to #16

### CODE REVIEW

File: docker/sweepers_driver.py [hunk 1]
```diff
@@ -58,11 +58,22 @@
...
+log = logging.getLogger(__name__)
+
+dev_mode = str(os.environ.get("DEV_MODE")).lower() not in {'none', '', '0', 'false'}
+if dev_mode:
+    log.warning(f'Operating in development mode - host verification disabled')
+    import urllib3
+    urllib3.disable_warnings()
 
 opensearch_endpoint = os.environ.get("PROV_ENDPOINT")
+log.info(f'Targeting base OpenSearch endpoint "{opensearch_endpoint}"')
```
Comment (nutjob4life) [inline docker/sweepers_driver.py:76]: Is there a reasonable default for `PROV_ENDPOINT` we can add as a second argument to `os.environ.get` in the case someone forgets to set the env var?

File: docker/sweepers_driver.py [hunk 2]
```diff
@@ -80,21 +91,44 @@ def run_factory(sweeper_f: Callable) -> Callable:
...
+
 run_provenance = run_factory(provenance.run)
 run_ancestry = run_factory(ancestry.run)
 
-cross_cluster_remote_batches = os.environ.get("PROV_REMOTES") or None
-if cross_cluster_remote_batches is None:
+cross_cluster_remote_node_batches = parse_cross_cluster_remotes(os.environ.get("PROV_REMOTES"))
+log.info(f'Running sweepers')
+if cross_cluster_remote_node_batches is None:
+    log.info(f'No CCS remotes specified - running sweepers against base OpenSearch endpoint only')
```
Comment (nutjob4life) [inline docker/sweepers_driver.py:122]: No need for an `f` string here?

File: docker/sweepers_driver.py [hunk 3]
```diff
@@ -80,21 +91,44 @@ def run_factory(sweeper_f: Callable) -> Callable:
...
     run_provenance()
     run_ancestry()
 else:
-    remote_nodesets = json.loads(cross_cluster_remote_batches)
-    for remote_nodeset in remote_nodesets:
-        cross_cluster_remotes = remote_nodeset.split()
-
+    f'CCS remotes specified: {json.dumps(cross_cluster_remote_node_batches)}'
```
Comment (nutjob4life) [inline docker/sweepers_driver.py:126]: What's this string doing here? We missing a `log.info(…)`?

File: docker/sweepers_driver.py (hunk 1 above)
Comment (nutjob4life) [inline docker/sweepers_driver.py:76]: When using Python logging, it's better to say `log.info('Blah blah %s to %s', thing1, thing2)` than to use string interpolation. If the `info` level is turned _off_, then no string substitutions are even executed.

Doing the string interpolation before the logging means it _always_ happens, even if that level is disabled.

File: docker/sweepers_driver.py [hunk 4]
```diff
@@ -80,21 +91,44 @@ def run_factory(sweeper_f: Callable) -> Callable:
...
+    for cross_cluster_remotes in cross_cluster_remote_node_batches:
+        targets_msg_str = f'base OpenSearch and the following remotes: {json.dumps(cross_cluster_remotes)}'
+        log.info(f'Running sweepers against {targets_msg_str}')
```
Comment (nutjob4life) [inline docker/sweepers_driver.py:129]: `log.info('Running sweepers against %s', targets_msg_str)`

File: docker/sweepers_driver.py [hunk 5]
```diff
@@ -80,21 +91,44 @@ def run_factory(sweeper_f: Callable) -> Callable:
...
         run_provenance(cross_cluster_remotes=cross_cluster_remotes)
         run_ancestry(cross_cluster_remotes=cross_cluster_remotes)
+        log.info(f'Successfully ran sweepers against base OpenSearch and {targets_msg_str}')
```
Comment (nutjob4life) [inline docker/sweepers_driver.py:132]: ʺ

File: src/pds/registrysweepers/utils/__init__.py [hunk 6]
```diff
@@ -112,7 +112,7 @@ def query_registry_db(
         "size": page_size,
     }
 
-    log.debug(f"Initiating query: {req_content}")
+    log.info(f"Initiating query: {req_content}")
```
Comment (nutjob4life) [inline src/pds/registrysweepers/utils/__init__.py:115]: ʺ

File: src/pds/registrysweepers/utils/__init__.py [hunk 7]
```diff
@@ -133,11 +133,21 @@ def query_registry_db(
         req_content = {"scroll": f"{scroll_validity_duration_minutes}m", "scroll_id": data["_scroll_id"]}
 
         total_hits = data["hits"]["total"]["value"]
-        log.info(f"   paging query ({served_hits} to {min(served_hits + page_size, total_hits)} of {total_hits})")
+        log.debug(f"   paging query ({served_hits} to {min(served_hits + page_size, total_hits)} of {total_hits})")
```
Comment (nutjob4life) [inline src/pds/registrysweepers/utils/__init__.py:136]: ʺ

File: src/pds/registrysweepers/utils/__init__.py [hunk 8]
```diff
@@ -133,11 +133,21 @@ def query_registry_db(
...
+        log.info("Query progress: 0%")
+
         for hit in data["hits"]["hits"]:
-            yield hit
             served_hits += 1
 
+            percentage_of_hits_served = int(served_hits / total_hits * 100)
+            if last_info_log_at_percentage is None or percentage_of_hits_served > (last_info_log_at_percentage + 5):
+                last_info_log_at_percentage = percentage_of_hits_served
+                log.info(f"Query progress: {percentage_of_hits_served}%")
```
Comment (nutjob4life) [inline src/pds/registrysweepers/utils/__init__.py:147]: ʺ

**Review by nutjob4life — CHANGES_REQUESTED** [review]
Thanks in advance!

File: docker/sweepers_driver.py (hunk 1 above)
Comment (alexdunnjpl) [reply docker/sweepers_driver.py:76]: Two possible approaches:
 - throw if not set
 - default to default local-dev values for endpoint and creds
 
 Thoughts on preference?

File: docker/sweepers_driver.py (hunk 2 above)
Comment (alexdunnjpl) [reply docker/sweepers_driver.py:122]: Will fix - previously I was printing the base URL but later determined it wasn't useful

File: docker/sweepers_driver.py (hunk 3 above)
Comment (alexdunnjpl) [reply docker/sweepers_driver.py:126]: Yanoit 😰

File: docker/sweepers_driver.py (hunk 1 above)
Comment (alexdunnjpl) [reply docker/sweepers_driver.py:76]: Agreed in principle, but seems like the kind of thing you that's only beneficial in performance-intensive contexts and outside of that I prefer to write for the human (readability is hardly an issue with a single substitution, but I still think it scans better).

Valid/not?

Comment (alexdunnjpl) [reply docker/sweepers_driver.py:76]: [I guess this is the best-of-both-worlds answer](https://discuss.python.org/t/safer-logging-methods-for-f-strings-and-new-style-formatting/13802#big-bonus-switching-logging-to-the-modern-formatting-style-3)

File: docker/sweepers_driver.py [hunk 9]
```diff
@@ -58,11 +58,22 @@
 import json
 import logging
 import os
-from typing import Callable
+from typing import Callable, Iterable
 
 from pds.registrysweepers import provenance, ancestry
+from pds.registrysweepers.utils import configure_logging
```
[the remark this answers is not in the document]
Comment (alexdunnjpl) [reply docker/sweepers_driver.py:64]: @sonatype-lift ignore

File: docker/sweepers_driver.py (hunk 1 above)
Comment (nutjob4life) [reply docker/sweepers_driver.py:76]: I'd throw. Fail fast!

Comment (nutjob4life) [reply docker/sweepers_driver.py:76]: No doubt f-strings read 1000% better. But how can you ever predict that your "this'll never be called in a loop" won't be called in the future in some kind of huge or even massively-parallel loop?

Using the ugly `%` subs for logging is, sadly, the standard practice—unless you do this:

https://docs.python.org/3/howto/logging-cookbook.html#formatting-styles

Comment (nutjob4life) [reply docker/sweepers_driver.py:76]: Never mind, this is fine

**Review by nutjob4life — APPROVED** [review]

**Pull request merged**